packages/suite/src/types/suite/index.ts

```diff
@@ -76,6 +76,9 @@ export type Action =
 export interface Dispatch extends ThunkDispatch<AppState, any, Action> {
     <Action>(action: Action): Action extends (...args: any) => infer R ? R : Action;
 }
+
+export declare type PromiseType<P extends Promise<any>> = P extends Promise<infer T> ? T : never;
```

Comment:
Utility types are grouped in `src/types/utils` so I'd move it there maybe?

Reply:
And also check `Await` type in `src/types/utils` which seems to be doing the same maybe?